frontend/src/features/map_planning/components/timeline/TimelineDatePicker.tsx

```diff
@@ -0,0 +1,301 @@
+
+  const handleDaySliderLeftEndReached = () => {
+    if (visibleDays[0].key > 0) {
+      setVisibleDays(calculateVisibleDays(selectedDayItem));
+    }
+  };
+
+  const handleDaySliderRightEndReached = () => {
+    if (visibleDays[visibleDays.length - 1].key < daySilderItems.length - 1) {
+      setVisibleDays(calculateVisibleDays(selectedDayItem));
+    }
+  };
+
+  useEffect(() => {
+    const timeoutId = setTimeout(() => {
+      const newDay = selectedDayItem;
+      onSelectDate(`${newDay.year}-${newDay.month}-${newDay.day}`);
+    }, 500);
+    return () => clearTimeout(timeoutId);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedDayItem]);
+
+  return (
+    <>
+      <ItemSliderPicker
+        items={yearlyTimeLineEvents.map((yearSliderItem) => ({
+          key: yearSliderItem.key,
+          content: TimelineDatePickerItem({
+            text: yearSliderItem.year.toString(),
+            added: yearSliderItem.added,
+            removed: yearSliderItem.removed,
+          }),
+        }))}
+        onChange={handleYearChange}
+        value={selectedYearItem.key}
+      />
+      <ItemSliderPicker
+        items={visibleMonths.map((monthSliderItem) => ({
+          key: monthSliderItem.key,
+          content: TimelineDatePickerItem({
+            text: getShortMonthNameFromNumber(monthSliderItem.month, i18n.resolvedLanguage),
+            added: monthSliderItem.added,
+            removed: monthSliderItem.removed,
+            disabled: monthSliderItem.year !== selectedYearItem.year,
+          }),
+        }))}
+        onChange={handleMonthChange}
+        leftEndReached={handleMontSliderLeftEndReached}
+        rightEndReached={handleMontSliderRightEndReached}
+        value={selectedMonthItem.key}
+      />
+      <ItemSliderPicker
+        testDataId={TEST_IDS.DAY_SLIDER}
+        items={visibleDays.map((daySliderItem) => ({
+          key: daySliderItem.key,
+          content: TimelineDatePickerItem({
+            text: daySliderItem.day.toString(),
+            added: daySliderItem.added,
+            removed: daySliderItem.removed,
+            disabled:
+              daySliderItem.month != selectedMonthItem.month ||
+              daySliderItem.year != selectedMonthItem.year,
+          }),
+        }))}
+        onChange={handleDayItemChange}
+        leftEndReached={handleDaySliderLeftEndReached}
+        rightEndReached={handleDaySliderRightEndReached}
+        value={selectedDayItem.key}
+      />
+    </>
+  );
+};
+
+function TimelineDatePickerItem({
+  text,
+  added,
+  removed,
+  disabled,
+}: {
+  text: string;
+  added: number;
+  removed: number;
+  disabled?: boolean;
+}) {
+  return (
+    <div>
+      <TimeLineDatePickerEventIndivatior added={added} removed={removed} />
+      <span className={`select-none ${disabled ? 'text-gray-400' : ''}`}>{text}</span>
+    </div>
+  );
+}
+
+function TimeLineDatePickerEventIndivatior({ added, removed }: { added: number; removed: number }) {
```

Comment:
Did you mean indicator instead of indivatior?
```suggestion
function TimeLineDatePickerEventIndicator({ added, removed }: { added: number; removed: number }) {
```